src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs

```diff
@@ -28,23 +28,92 @@ public DependencyContext Read(Stream stream)
 
         private DependencyContext Read(JObject root)
         {
+            string runtime = string.Empty;
+            string target = string.Empty;
+
+            var runtimeTargetInfo = ReadRuntimeTargetInfo(root);
             var libraryStubs = ReadLibraryStubs((JObject) root[DependencyContextStrings.LibrariesPropertyName]);
             var targetsObject = (IEnumerable<KeyValuePair<string, JToken>>) root[DependencyContextStrings.TargetsPropertyName];
 
-            var runtimeTargetProperty = targetsObject.First(target => IsRuntimeTarget(target.Key));
-            var compileTargetProperty = targetsObject.First(target => !IsRuntimeTarget(target.Key));
+            JObject runtimeTarget = null;
+            JObject compileTarget = null;
+            if (targetsObject != null)
+            {
+                var compileTargetProperty = targetsObject.FirstOrDefault(t => !IsRuntimeTarget(t.Key));
+                compileTarget = (JObject) compileTargetProperty.Value;
+                target = compileTargetProperty.Key;
+
+                if (!string.IsNullOrEmpty(runtimeTargetInfo.Name))
```

Comment:
I wonder if maybe we should put the compilation target name in as well just for completeness... that way nobody is parsing target names. Not sure if that might be overkill though.